### PR TITLE
Enforce a reason is provided for revert command

### PR DESCRIPTION
**Pull request by janeyx99**
Enforce that people will put a revert reason when reverting through GH1.

note: there are lots of lint changes here cuz I autoformatted with vs code :/

### CODE REVIEW

File: torchci/lib/bot/mergeBot.ts [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import { Probot } from "probot";
-import { reactOnComment } from './botUtils'
+import { addComment, reactOnComment } from "./botUtils";
 
 function mergeBot(app: Probot): void {
-  const mergeCmdPat = new RegExp("^\\s*@pytorch(merge|)bot\\s+(force\\s+)?merge\\s+this");
+  const mergeCmdPat = new RegExp(
```
Comment (janeyx99) [inline torchci/lib/bot/mergeBot.ts:5]: lint change :/

File: torchci/lib/bot/mergeBot.ts [hunk 2]
```diff
@@ -12,14 +14,16 @@ function mergeBot(app: Probot): void {
...
     async function dispatchEvent(event_type: string, force: boolean = false) {
-      let payload = force ? {
-        pr_num: prNum,
-        comment_id: ctx.payload.comment.id,
-        force: true,
-      } : {
-        pr_num: prNum,
-        comment_id: ctx.payload.comment.id,
-      };
+      let payload = force
```
Comment (janeyx99) [inline torchci/lib/bot/mergeBot.ts:17]: lint change :/

File: torchci/lib/bot/mergeBot.ts [hunk 3]
```diff
@@ -81,11 +97,12 @@ function mergeBot(app: Probot): void {
       });
     }
 
-    if (reviewBody?.match(mergeCmdPat)) {
-      await dispatchEvent("try-merge");
-      await addComment("+1"); // REST API doesn't support reactions for code reviews.
+      if (reviewBody?.match(mergeCmdPat)) {
```
Comment (janeyx99) [inline torchci/lib/bot/mergeBot.ts:100]: lint change :/

File: torchci/test/mergeBot.test.ts [hunk 4]
```diff
@@ -64,7 +64,6 @@ describe("merge-bot", () => {
     scope.done();
   });
 
-
```
Comment (janeyx99) [inline torchci/test/mergeBot.test.ts:67]: lint change :/

File: torchci/test/mergeBot.test.ts [hunk 5]
```diff
@@ -76,9 +75,7 @@ describe("merge-bot", () => {
       .post(
         `/repos/${owner}/${repo}/issues/comments/${comment_number}/reactions`,
         (body) => {
-          expect(JSON.stringify(body)).toContain(
-            "{\"content\":\"confused\"}"
-          );
+          expect(JSON.stringify(body)).toContain('{"content":"confused"}');
```
Comment (janeyx99) [inline torchci/test/mergeBot.test.ts:78]: lint change :/

File: torchci/test/mergeBot.test.ts [hunk 6]
```diff
@@ -104,22 +101,17 @@ describe("merge-bot", () => {
       .post(
         `/repos/${owner}/${repo}/issues/comments/${comment_number}/reactions`,
         (body) => {
-          expect(JSON.stringify(body)).toContain(
-            "{\"content\":\"+1\"}"
-          );
```
Comment (janeyx99) [inline torchci/test/mergeBot.test.ts:109]: lint change :/

File: torchci/test/mergeBot.test.ts [hunk 7]
```diff
@@ -104,22 +101,17 @@ describe("merge-bot", () => {
...
+          expect(JSON.stringify(body)).toContain('{"content":"+1"}');
           return true;
         }
       )
       .reply(200, {})
-      .post(
```
Comment (janeyx99) [inline torchci/test/mergeBot.test.ts:114]: lint change :/

File: torchci/test/mergeBot.test.ts [hunk 8]
```diff
@@ -142,22 +134,17 @@ describe("merge-bot", () => {
       .post(
         `/repos/${owner}/${repo}/issues/comments/${comment_number}/reactions`,
         (body) => {
-          expect(JSON.stringify(body)).toContain(
```
Comment (janeyx99) [inline torchci/test/mergeBot.test.ts:145]: lint change :/

File: torchci/test/mergeBot.test.ts [hunk 9]
```diff
@@ -180,22 +194,17 @@ describe("merge-bot", () => {
...
```
Comment (janeyx99) [inline torchci/test/mergeBot.test.ts:183]: lint change :/

File: torchci/test/mergeBot.test.ts [hunk 10]
```diff
@@ -180,22 +194,17 @@ describe("merge-bot", () => {
...
-            "{\"content\":\"+1\"}"
-          );
+          expect(JSON.stringify(body)).toContain('{"content":"+1"}');
           return true;
         }
       )
       .reply(200, {})
-      .post(
```
Comment (janeyx99) [inline torchci/test/mergeBot.test.ts:190]: lint change :/

File: torchci/test/mergeBot.test.ts [hunk 11]
```diff
@@ -254,25 +262,17 @@ describe("merge-bot", () => {
     const repo = event.payload.repository.name;
     const pr_number = event.payload.pull_request.number;
     const scope = nock("https://api.github.com")
-      .post(
```
Comment (janeyx99) [inline torchci/test/mergeBot.test.ts:257]: lint change :/

File: torchci/test/mergeBot.test.ts [hunk 12]
```diff
@@ -254,25 +262,17 @@ describe("merge-bot", () => {
...
-          );
-          return true;
-        }
-      )
+      .post(`/repos/${owner}/${repo}/issues/${pr_number}/comments`, (body) => {
+        expect(JSON.stringify(body)).toContain('{"body":"+1"}');
+        return true;
+      })
       .reply(200, {})
-      .post(
```
Comment (janeyx99) [inline torchci/test/mergeBot.test.ts:267]: lint change :/

File: torchci/lib/bot/mergeBot.ts [hunk 13]
```diff
@@ -44,6 +48,18 @@ function mergeBot(app: Probot): void {
...
         return;
       }
+      const revertWithReasonCmdPat = new RegExp(
+        "^\\s*@pytorch(merge|)bot\\s+revert\\s+this.*(\\s+\\w+){3,}"
+      );
+      if (commentBody.match(revertWithReasonCmdPat) === null) {
+        // revert reason of 3+ words not given
+        await addComment(
+          ctx,
+          "Revert unsuccessful: please retry the command explaining why the revert is necessary, " +
```
Comment (zengk95) [inline torchci/lib/bot/mergeBot.ts:58]: might want something for 3+ words but guess it's fine. Or have them do like reason=blah so we can more easily parse out reasons down the line.

**Review by zengk95 — APPROVED** [review]
LGTM

File: torchci/lib/bot/mergeBot.ts (hunk 13 above)
Comment (janeyx99) [reply torchci/lib/bot/mergeBot.ts:54]: Looking at the revert commands from the last 2 weeks, it looks like people have already been giving revert reasons in their own way so I don't think it'll be all that worth to switch it up and use reason=. We can just take the whole comment and it's usually good. https://fburl.com/scuba/opensource_github_webhook_events/p7fylx1x

**Pull request merged**